### ISSUE TITLE
Build fail 3.2.1

### ISSUE THREAD

**Issue by kyonge** [danielgindi/ChartsRealm#27]
Hi, I want to use ChartsRealm so I tried to build with Carthage.
But build was failed as the log below.

```
Kiui-MacBook-Pro:StoreData_iOS kiyonglee$ carthage update --platform iOS
*** Fetching ChartsRealm
*** Fetching Charts
*** Fetching realm-cocoa
*** Checking out ChartsRealm at "3.2.1"
*** Checking out realm-cocoa at "v3.13.1"
*** Checking out Charts at "v3.2.2"
*** xcodebuild output can be found in /var/folders/ld/2v13t6_x5vg8vfy2jr7d_j680000gn/T/carthage-xcodebuild.RHVNLX.log
*** Downloading realm-cocoa.framework binary at "v3.13.1"
*** Building scheme "Charts" in Charts.xcworkspace
*** Building scheme "ChartsRealm" in ChartsRealm.xcodeproj
Build Failed
	Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -project /Users/kiyonglee/Desktop/Contents/Dev/OUB/StoreData_iOS/Carthage/Checkouts/ChartsRealm/ChartsRealm.xcodeproj -scheme ChartsRealm -configuration Release -derivedDataPath /Users/kiyonglee/Library/Caches/org.carthage.CarthageKit/DerivedData/10.1_10B61/ChartsRealm/3.2.1 -sdk iphoneos ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive -archivePath /var/folders/ld/2v13t6_x5vg8vfy2jr7d_j680000gn/T/ChartsRealm SKIP_INSTALL=YES GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO CLANG_ENABLE_CODE_COVERAGE=NO STRIP_INSTALLED_PRODUCT=NO (launched in /Users/kiyonglee/Desktop/Contents/Dev/OUB/StoreData_iOS/Carthage/Checkouts/ChartsRealm)

This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/ld/2v13t6_x5vg8vfy2jr7d_j680000gn/T/carthage-xcodebuild.RHVNLX.log
```

```
...
/Users/kiyonglee/Desktop/Contents/Dev/OUB/StoreData_iOS/Carthage/Checkouts/ChartsRealm/ChartsRealm/Classes/Data/RealmPieDataSet.swift:19:12: error: type 'RealmPieDataSet' does not conform to protocol 'IP$
open class RealmPieDataSet: RealmBaseDataSet, IPieChartDataSet
           ^
Charts.IPieChartDataSet:8:16: note: protocol requires property 'useValueColorForLine' with type 'Bool'; do you want to add a stub?
    public var useValueColorForLine: Bool { get set }
               ^
/Users/kiyonglee/Desktop/Contents/Dev/OUB/StoreData_iOS/Carthage/Checkouts/ChartsRealm/ChartsRealm/Classes/Data/RealmBaseDataSet.swift:630:20: error: static member 'copyWithZone' cannot be used on instan$
        let copy = super.copyWithZone(zone) as! RealmBaseDataSet
                   ^     ~~~~~~~~~~~~


** ARCHIVE FAILED **


The following build commands failed:
        CompileSwift normal armv7
        CompileSwiftSources normal armv7 com.apple.xcode.tools.swift.compiler
        CompileSwift normal arm64
        CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(4 failures)
```

The method `copyWithZone` is replaced to `copy(with zone: NSZone? = nil) -> Any` in Swift4.
Would you fix this issue in next update please?

Thank you so much.

**Comment by filipetamota** [comment]
The same problem exists with pods

**Comment by liuxuan30** [comment]
>Update to Xcode 10 and Swift 4.2
Fix some warnings and self-calling init()
Charts update to 3.2.1

I tested and build with Swift 4.2 when updating to pod. Why it's failing here

another option is you pick up the source code, not using precompiled one. Or you stick to the same version as what I released.

I checked your log, it seems you are upgrading Charts and Realm, but what I built is against 3.2.1
>*** Checking out realm-cocoa at "v3.13.1"
*** Checking out Charts at "v3.2.2"
>Charts.IPieChartDataSet:8:16: note: protocol requires property 'useValueColorForLine' with type 'Bool'; do you want to add a stub?
    public var useValueColorForLine: Bool { get set }


anyway, are you able to file a PR to fix it? I don't see this is a hot project and then it has a low priority. My current focus is on Charts itself. I'd love to fix issues for ChartsRealm, but that have to wait until then. So I may need your help to fix this.
